pkg/smokescreen/smokescreen.go

```diff
@@ -852,13 +890,14 @@ func getRole(config *Config, req *http.Request) (string, error) {
 	}
 }
 
-func checkIfRequestShouldBeProxied(config *Config, req *http.Request, outboundHost string) (*aclDecision, time.Duration, error) {
-	decision := checkACLsForRequest(config, req, outboundHost)
+func checkIfRequestShouldBeProxied(config *Config, req *http.Request, host string, port int) (*aclDecision, time.Duration, error) {
+	decision := checkACLsForRequest(config, req, host, port)
 
 	var lookupTime time.Duration
 	if decision.allow {
 		start := time.Now()
-		resolved, reason, err := safeResolve(config, "tcp", outboundHost)
+		hostPort := net.JoinHostPort(host, strconv.Itoa(port))
+		resolved, reason, err := safeResolve(config, "tcp", hostPort)
```

Comment:
DNS resolution also uses the normalized host

Reply:
Correct!